pkg/validate/dynamic/platformworkloadidentityprofile.go

```diff
@@ -0,0 +1,117 @@
+package dynamic
+
+import (
+	"context"
+	"net/http"
+
+	sdkauthorization "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v3"
+	"github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/armauthorization"
+	"github.com/Azure/ARO-RP/pkg/util/rbac"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func (dv *dynamic) ValidatePlatformWorkloadIdentityProfile(ctx context.Context, oc *api.OpenShiftCluster, platformWorkloadIdentityRoles []api.PlatformWorkloadIdentityRole, roleDefinitions armauthorization.RoleDefinitionsClient) error {
+	dv.log.Print("ValidatePlatformWorkloadIdentityProfile")
+
+	platformIdentitiesActionsMap := map[string][]string{}
+
+	for _, role := range oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+		platformIdentitiesActionsMap[role.OperatorName] = nil
+	}
+
+	for _, role := range platformWorkloadIdentityRoles {
+		_, ok := platformIdentitiesActionsMap[role.OperatorName]
+		if !ok {
+			requiredOperatorIdentities := []string{}
+			for _, role := range platformWorkloadIdentityRoles {
```

Comment:
You could rename this loop variable to something other than `role`, since it shadows the `role` variable declared for the outer loop and makes the code a little confusing to read.

Or alternatively you could do something like this:

```
roleIsMissing := false
requiredOperatorIdentities := []string{}
for _, role := range platformWorkloadIdentityRoles {
    requiredOperatorIdentities = append(requiredOperatorIdentities, role.OperatorName)
    if role not in platformIdentitiesActionsMap {
        roleIsMissing = true
    }
}

if roleIsMissing || len(platformWorkloadIdentityRoles) != len(platformIdentitiesActionsMap) {
    return the error
}
```

Note that I'm also checking the length of the cx's role set, which I think your original implementation is missing; in addition to making sure they haven't missed any, we want to make sure they aren't providing extras either.

Reply:
Thanks for providing the snippet, used it as it is. Yes, the length check was added in the local branch, but used your provided approach.